drivers/counter/counter_ace_v1x_rtc.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <devicetree.h>
+#include <zephyr/drivers/counter.h>
+#include <soc.h>
+#include <ace_v1x-regs.h>
+#include <counter/counter_ace_v1x_rtc_regs.h>
+
+#if defined(__XCC__)
+#include <xtensa/config/core.h>
+#if XCHAL_HAVE_HIFI4 && (defined(__ALDERLAKE__) \
+	|| defined(__METEORLAKE__) || defined(__METEORLAKE_S__))
```

Comment:
No need for platform macros here.  If the toolchain exposes support for HiFi, then it's legal to include the header.